addons/website/models/website.py

```diff
@@ -187,6 +187,15 @@ def _compute_menu(self):
             top_menus = menus.filtered(lambda m: not m.parent_id)
             website.menu_id = top_menus and top_menus[0].id or False
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_default_website(self):
+        try:
+            default_website = self.env.ref('website.default_website')
+        except ValueError:
+            default_website = None
+        if default_website and default_website in self:
+            raise UserError("You can't delete '%s', as it is system generated" % (default_website.name))
```

Comment:
The message could be improved, e.g.
```suggestion
            raise UserError(_("You cannot delete default website %s. Try to change its settings instead", default_website.name))
```